engines/nancy/action/telephone.cpp

```diff
@@ -0,0 +1,316 @@
+
+    NancySceneState.setShouldClearTextbox(false);
+}
+
+void Telephone::readData(Common::SeekableReadStream &stream) {
+    char buf[10];
+    stream.read(buf, 10);
+    imageName = buf;
+
+    for (uint i = 0; i < 12; ++i) {
+        srcRects.push_back(Common::Rect());
+        readRect(stream, srcRects.back());
+    }
+
+    for (uint i = 0; i < 12; ++i) {
+        destRects.push_back(Common::Rect());
+        readRect(stream, destRects.back());
+
+        if (i == 0) {
+            _screenPosition = destRects.back();
+        } else {
+            _screenPosition.extend(destRects.back());
+        }
+    }
+
+    genericDialogueSound.read(stream, SoundDescription::kNormal);
+    genericButtonSound.read(stream, SoundDescription::kNormal);
+    ringSound.read(stream, SoundDescription::kNormal);
+    dialToneSound.read(stream, SoundDescription::kNormal);
+    dialAgainSound.read(stream, SoundDescription::kNormal);
+    hangUpSound.read(stream, SoundDescription::kNormal);
+
+    for (uint i = 0; i < 12; ++i) {
+        stream.read(buf, 10);
+        buttonSoundNames.push_back(buf);
+    }
+
+    char buf2[200];
+    stream.read(buf2, 200);
+    addressBookString = buf2;
+    stream.read(buf2, 200);
+    dialAgainString = buf2;
+    reloadScene.readData(stream);
+    stream.skip(2);
+    flagOnReload.label = stream.readSint16LE();
+    flagOnReload.flag = (NancyFlag)stream.readUint16LE();
+    exitScene.readData(stream);
+    stream.skip(2);
+    flagOnExit.label = stream.readSint16LE();
+    flagOnExit.flag = (NancyFlag)stream.readUint16LE();
+    readRect(stream, exitHotspot);
+
+    uint numCalls = stream.readUint16LE();
+
+    for (uint i = 0; i < numCalls; ++i) {
+        calls.push_back(PhoneCall());
+        PhoneCall &call = calls.back();
+
+        for (uint j = 0; j < 11; ++j) {
+            call.phoneNumber.push_back(stream.readByte());
+        }
+        
+        stream.read(buf, 10);
+        call.soundName = buf;
+        stream.read(buf2, 200);
+        call.text = buf2;
+        call.sceneChange.readData(stream);
+        stream.skip(2);
+        call.flag.label = stream.readSint16LE();
+        call.flag.flag = (NancyFlag)stream.readUint16LE();
+    }
+}
+
+void Telephone::execute() {
+    switch (state) {
+    case kBegin:
+        init();
+        registerGraphics();
+        g_nancy->sound->loadSound(dialToneSound);
+        g_nancy->sound->playSound(dialToneSound);
+        NancySceneState.getTextbox().clear();
+        NancySceneState.getTextbox().addTextLine(addressBookString);
+        state = kRun;
+        // fall through
+    case kRun:
+        switch (callState) {
+        case kWaiting:
+            // Long phone numbers start with 1
+            if (calledNumber.size() >= 11 || (calledNumber.size() >= 7 && (calledNumber[0] != 1))) {
+                NancySceneState.getTextbox().clear();
+                NancySceneState.getTextbox().addTextLine("ringing...<n><e>"); // Hardcoded in the original engine
```

Comment:
Not sure if the original game supports different languages - but this should get marked as translatable maybe and a potfiles should get added to the engine (if not already there)

Reply:
Done.